_example/db/pagination.go

```diff
@@ -24,6 +25,10 @@ type Pagination struct {
 }
 
 func (p *Pagination) Paginate(c *gin.Context) (*gorm.DB, error) {
+	if p == nil {
+		return nil, errors.New("Pagination struct got nil.")
+	}
```

Comment:
Is this guard really needed?

I think nil check should be done before calling any struct method.